graphql/resolver.go

```diff
@@ -35,6 +36,54 @@ func New(apiURL string) Config {
 			sc: dbConnector,
 		},
 	}
+	c.Directives.RequireHostAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver, access HostAccessLevel) (interface{}, error) {
+		args, isStringMap := obj.(map[string]interface{})
+		if !isStringMap {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostId, hasHostId := args["hostId"].(string)
+		hostIdsInterface, hasHostIds := args["hostIds"].([]interface{})
+		if !hasHostId && !hasHostIds {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+
+		hostIds := []string{}
+		if hasHostIds {
+			for _, v := range hostIdsInterface {
+				hostIds = append(hostIds, v.(string))
+			}
+		}
+		hostIdsToCheck := []string{hostId}
+		hostIdsToCheck = append(hostIdsToCheck, hostIds...)
+		var requiredLevel int
+		if access == HostAccessLevelEdit {
+			requiredLevel = evergreen.HostsEdit.Value
+		} else {
+			requiredLevel = evergreen.HostsView.Value
+		}
+		user := mustHaveUser(ctx)
+		hostsToCheck, err := host.Find(ctx, host.ByIds(hostIdsToCheck))
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, "Error getting hosts")
+		}
+		if len(hostsToCheck) == 0 {
+			return nil, ResourceNotFound.Send(ctx, "No matching hosts found")
+		}
+		forbiddenHosts := []string{}
+		for _, h := range hostsToCheck {
+			if !userHasHostPermission(user, h.Distro.Id, requiredLevel) {
+				forbiddenHosts = append(forbiddenHosts, h.Id)
+			}
+		}
+		if len(forbiddenHosts) == 1 {
+			return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access host '%v'", user.Username(), forbiddenHosts[0]))
+		} else if len(forbiddenHosts) > 1 {
+			hostsString := strings.Join(forbiddenHosts, ", ")
+			return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access hosts: '%v'", user.Username(), hostsString))
+		}
```

Comment:
could this be consolidated? I also thing usink `%s` might be slightly better than using `%v` since it's a string
```suggestion
		if len(forbiddenHosts) > 0 {
			hostsString := strings.Join(forbiddenHosts, ", ")
			return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access host(s): '%s'", user.Username(), hostsString))
		}
```

Reply:
This could be consolidated but I like how the copy is more specific in the original code. I updated %v to %s.